lib/kernel/doc/src/pg.xml

```diff
@@ -133,26 +222,53 @@
     <func>
       <name name="join" arity="2" since="OTP 23.0"/>
       <name name="join" arity="3" since="OTP 23.0"/>
-      <fsummary>Join a process or a list of processes to a group.</fsummary>
+      <name name="join" arity="4" since="OTP 27.0"/>
+      <fsummary>Join a member or a list of members to a group.</fsummary>
       <desc>
-        <p>Joins single process or multiple processes to the
-          group <c>Group</c>. A process can join a group many times and
+        <p>If <c>ShardRanges</c> is not specified, joins single member or
+          multiple members <c>MemberOrMembers</c> to the singleton group
+          <c>Group</c>. If <c>ShardRanges</c> is specified, joins
+          <c>MemberOrMembers</c> to the <c>ShardRanges</c> of the sharded
+          group <c>Group</c>. A member can join a group many times and
           must then leave the group the same number of times.</p>
-        <p><c>PidOrPids</c> may contain the same process multiple times.</p>
+        <p><c>MemberOrMembers</c> may contain the same member multiple times.
+          </p>
       </desc>
     </func>
 
     <func>
       <name name="leave" arity="2" since="OTP 23.0"/>
       <name name="leave" arity="3" since="OTP 23.0"/>
-      <fsummary>Make a process leave a group.</fsummary>
+      <name name="leave" arity="4" since="OTP 27.0"/>
+      <fsummary>Make a member or a list of members leave a group.</fsummary>
+      <desc>
```

Comment:
The leave behavior is a bit complex.
In current pg, ```leave``` returns ```not_joined``` only if ALL leave requests doesn't success. If any of the pid in the list was joined, it will return ok.
This behavior I think is not good, or even bug-like, although documented. If the server can handle unexpected leave, it should always return ok. If it cannot, it should fail if ANY pid was not joined.

When in shard case the behavior will be more confusing to user, what if you only joined ```{2, 4}``` and then leave ```{3, 6}```? Should it leave ```{3, 4}``` or completely not leave? I think the server should reply what was not expected and do not execute the whole operation at all (especially with ```update()``` function). That's what I'm currently implementing.

Any choice here should be implementable, although the "let unexpected leave silently work but only return (a simple) error when non of them success" will introduce a lot of code. What do you suggest?